roles/manifest/tasks/download_manifest.yml

```diff
@@ -0,0 +1,9 @@
+---
+- name: Download Subscription Manifest from Red Hat Customer Portal
+  get_url:
```

Comment:
we have a module for that :)

https://theforeman.github.io/foreman-ansible-modules/develop/plugins/redhat_manifest_module.html#ansible-collections-theforeman-foreman-redhat-manifest-module